packages/plugins/scripts-runner/src/index.ts

```diff
@@ -0,0 +1,178 @@
+import { eachSeries } from 'async';
+import { promises as fsPromises, stat, Stats } from 'fs';
+import path from 'path';
+import { Embark, Callback } from 'embark-core';
+import { Logger } from 'embark-logger';
+import { dappPath } from 'embark-utils';
+import Web3 from "web3";
+
+export enum ScriptsRunnerCommand {
+  Execute = 'scripts-runner:execute'
+}
+
+export interface ExecuteOptions {
+  env: string;
+}
+
+export interface ScriptDependencies {
+  web3: Web3 | null;
+  contracts: any;
+  logger: Logger;
+}
+
+enum FileType {
+  SymbolicLink = 'symbolic link',
+  Socket = 'socket',
+  Unknown = 'unknown'
+}
+
+class UnsupportedTargetError extends Error {
+
+  name = 'UnsupportedTargetError';
+
+  constructor(public stats: Stats) {
+    super();
+    // We can't access `this` before `super()` is called so we have to
+    // set `this.message` after that to get a dedicated error message.
+    this.setMessage();
+  }
+
+  private setMessage() {
+    let ftype = FileType.Unknown;
+    if (this.stats.isSymbolicLink()) {
+      ftype = FileType.SymbolicLink;
+    } else if (this.stats.isSocket()) {
+      ftype = FileType.Socket;
+    }
+    this.message = `Script execution target not supported. Expected file path or directory, got ${ftype} type.`;
+  }
+}
+
+class ScriptExecutionError extends Error {
+
+  name = 'ScriptExecutionError';
+
+  constructor(public target: string, public innerError: Error) {
+    super();
+    this.message = `Script '${path.basename(target)}' failed to execute. Original error: ${innerError}`;
+  }
+}
+
+export default class ScriptsRunnerPlugin {
+
+  embark: Embark;
+  private _web3: Web3 | null = null;
+
+  constructor(embark) {
+    this.embark = embark;
+    // TODO: it'd be wonderful if Embark called `registerCommandHandlers()` for us
+    this.registerCommandHandlers();
```

Comment:
This to do requires work in embark core as discussed many times in the past. Can be ignored for this PR.